app/events/proxies/proxy_deleted_event.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class Proxies::ProxyDeletedEvent < ProxyRelatedEvent
+
+  # @param [Cinstance] application
+  def self.create(proxy)
+    new(
+      proxy_id: proxy.id,
+      service_id: proxy.service_id,
+      metadata: {
+        provider_id: proxy.account.try(:id) || proxy.tenant_id
+      }
+    )
+  end
+
+  def self.find_or_build_for_service_id(service_id)
+    (EventStore::Event.where(event_type: Proxies::ProxyDeletedEvent).where("data LIKE '%service_id: #{service_id}%'").last || Proxies::ProxyDeletedEvent.new(data: {proxy_id: nil, service_id: service_id}))
```

Comment:
This should be tested in a similar environment to what we have in production, for that we thought in doing a query inside the preview DB, and not this exact query because as we don't have any `Proxies::ProxyDeletedEvent` yet, it would be faster than in real life, so do a similar query instead with an Event that we currently have in the DB, and that implies querying by event_type and by an attribute or its data.

A possible improvement would be querying only for this specific tenant as well 😄 

Right now we have the following indexes on this table:
https://github.com/3scale/porta/blob/e4e69790516bf223fee705887c3650a1e6710ee3/db/schema.rb#L485-L488
We are also considering that it may be faster if we add another index on `event_type` with `provider_id` 😄